build/gengo.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [[ -z "${STARTED_FROM_GENGO_BZL:-}" ]]; then
+  echo "Call this script from build/gengo.bzl"
+  exit 1
+fi
+
+# Main logic which runs the gengo tool on packages requesting generation.
+
+# Set DEBUG=1 for extra info and checks
+DEBUG=
+# location of the prebuilt generator
+dcg="$PWD/$tool"
```

Comment:
it might be more obvious that variables like `$tool` are set elsewhere if they're all-caps?